libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity.Editor/WebcamSourceEditor.cs

```diff
@@ -0,0 +1,525 @@
+                    {
+                        codecValue = SdpVideoCodecs.None;
+                    }
+                    else
+                    {
+                        try
+                        {
+                            codecValue = (SdpVideoCodecs)Enum.Parse(typeof(SdpVideoCodecs), _preferredVideoCodec.stringValue);
+                        }
+                        catch
+                        {
+                            codecValue = SdpVideoCodecs.Custom;
+                            customCodecValue = _preferredVideoCodec.stringValue;
+                            // Hide internal marker
+                            if (customCodecValue == "__CUSTOM")
+                            {
+                                customCodecValue = string.Empty;
+                            }
+                        }
+                    }
+
+                    // Display the edit field for the enum
+                    var newCodecValue = (SdpVideoCodecs)EditorGUI.EnumPopup(rect, _preferredVideoCodec.displayName, codecValue);
+                    if (newCodecValue == SdpVideoCodecs.H264)
+                    {
+                        EditorGUILayout.HelpBox("H.264 is only supported on UWP platforms.", MessageType.Warning);
+                    }
+
+                    // Update the value if changed or custom
+                    if ((newCodecValue != codecValue) || (newCodecValue == SdpVideoCodecs.Custom))
+                    {
+                        if (newCodecValue == SdpVideoCodecs.None)
+                        {
+                            _preferredVideoCodec.stringValue = string.Empty;
+                        }
+                        else if (newCodecValue == SdpVideoCodecs.Custom)
+                        {
+                            ++EditorGUI.indentLevel;
+                            string newValue = EditorGUILayout.TextField("SDP codec name", customCodecValue);
+                            if (newValue == string.Empty)
+                            {
+                                EditorGUILayout.HelpBox("The SDP codec name must be non-empty. See https://en.wikipedia.org/wiki/RTP_audio_video_profile for valid names.", MessageType.Error);
+
+                                // Force a non-empty value now, otherwise the field will reset to None
+                                newValue = "__CUSTOM";
+                            }
+                            _preferredVideoCodec.stringValue = newValue;
+                            --EditorGUI.indentLevel;
+                        }
+                        else
+                        {
+                            _preferredVideoCodec.stringValue = Enum.GetName(typeof(SdpVideoCodecs), newCodecValue);
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                    EditorGUILayout.PropertyField(_preferredVideoCodec);
```

Comment:
Document what's supposed to trigger this and what's the logic here.

Reply:
Removed.